src/addons/metaservice/src/main/java/ee/ria/xroad/proxy/serverproxy/RestMetadataServiceHandlerImpl.java

```diff
@@ -231,7 +233,21 @@ private void handleGetOpenApi(ProxyMessage requestProxyMessage) throws IOExcepti
                             serviceDescriptionURL, statusLine.getStatusCode(), statusLine.getReasonPhrase()));
         }
 
-        IOUtils.copy(response.getEntity().getContent(), restResponseBody);
+        InputStream responseContent = response.getEntity().getContent();
+
+        try {
+            OpenapiDescriptionFiletype filetype = getFileType(restResponse, serviceDescriptionURL);
```

Comment:
Shoud examine the http request `response`, not `restResponse`.

Reply:
Fixed.